tests/test_circulator_reports.py

```diff
@@ -1,103 +1,178 @@
-        ret = session.query(CirculatorArrival)
-        assert ret.count() > 100
 
+def test_factory(fake_session):
+    """Test for factory"""
+    ArrivalFactory()
+    ArrivalFactory(scheduled_arrival_time=time(12, 0))
 
-def test_get_otp_force(ridesystems_reports):
-    """Test for get_otp with the force option on existing data"""
-    ridesystems_reports.get_otp(date(2021, 5, 1), date(2021, 5, 1), force=True, hours='12')
-    with Session(bind=ridesystems_reports.engine, future=True) as session:
-        ret = session.query(CirculatorArrival)
-        assert ret.count() > 100
+    assert fake_session.query(CirculatorArrival).count() == 2
 
+@patch('transitstat.circulator.reports.RideSystemsInterface')
+def test_get_otp_no_force(mocked_rs_cls, conn_str):
+    """Test for get_otp"""
+    test_df = {
+        'date': [date(2022, 1, 1), date(2022, 1, 1), date(2022, 1, 1), date(2022, 1, 1)],
+        'route': ['xx', 'xx', 'xx', 'xx'],
+        'stop': ['xx', 'xx', 'xx', 'xx'],
+        'blockid': ['xx', 'xx', 'xx', 'xx'],
+        'actualarrivaltime': [time(), time(), time(), time()],
+        'actualdeparturetime': [time(), time(), time(), time()],
+        'scheduledarrivaltime': [time(), time(1), time(2), time(3)],
+        'scheduleddeparturetime': [time(), time(), time(), time()],
+        'vehicle': ['xx', 'xx', 'xx', 'xx'],
+        'ontimestatus': ['xx', 'xx', 'xx', 'xx']
+        }
+
+    inst = RidesystemReports(conn_str)
+    inst.rs_cls._login.return_value = "Login successful :)"
+    inst.rs_cls.get_otp.return_value = pd.DataFrame(test_df)
+
+    inst.get_otp(date(2022, 1, 1), date(2022, 1, 1), hours='12')
+    with Session(bind=inst.engine, future=True) as session:
+        ret = session.query(CirculatorArrival)
+        assert ret.count() == 8
 
-def test_get_otp_no_force(ridesystems_reports):
-    """Test for get_otp disabling the force option on existing data"""
-    ridesystems_reports.get_otp(date(2021, 5, 1), date(2021, 5, 1), force=False)
-    with Session(bind=ridesystems_reports.engine, future=True) as session:
+@patch('transitstat.circulator.reports.RideSystemsInterface')
+def test_get_otp_force(mocked_rs_cls, conn_str):
+    """Test for get_otp"""
+    test_df = {
+        'date': [date(2021, 5, 1), date(2021, 5, 2), date(2021, 5, 3), date(2021, 5, 4)],
+        'route': ['xx', 'xx', 'xx', 'xx'],
+        'stop': ['xx', 'xx', 'xx', 'xx'],
+        'blockid': ['xx', 'xx', 'xx', 'xx'],
+        'actualarrivaltime': [time(), time(), time(), time()],
+        'actualdeparturetime': [time(), time(), time(), time()],
+        'scheduledarrivaltime': [time(), time(), time(), time()],
+        'scheduleddeparturetime': [time(), time(), time(), time()],
+        'vehicle': ['1', '2', '3', '4'],
+        'ontimestatus': ['xx', 'xx', 'xx', 'xx']
+        }
+
+    inst = RidesystemReports(conn_str)
+    inst.rs_cls._login.return_value = "Login successful :)"
+    inst.rs_cls.get_otp.return_value = pd.DataFrame(test_df)
```

Comment:
All the same comments from above, but I also don't think this is doing the mocks properly, because inst.rs_cls isn't a mock is it?